rasa/nlu/utils/spacy_utils.py

```diff
@@ -70,82 +90,121 @@ def required_packages(cls) -> List[Text]:
+        return {
+            # when retrieving word vectors, this will decide if the casing
+            # of the word is relevant. E.g. `hello` and `Hello` will
+            # retrieve the same vector, if set to `False`. For some
+            # applications and models it makes sense to differentiate
+            # between these two words, therefore setting this to `True`.
+            "case_sensitive": False,
+        }
 
-    def doc_for_text(self, text: Text) -> "Doc":
-        """Makes a spaCy doc object from a string of text."""
-        return self.nlp(self.preprocess_text(text))
+    def __init__(
+        self, config: Dict[Text, Any]
+    ) -> None:
+        """Initializes a `SpacyPreprocessor`."""
+        self._config = config
+
+    @classmethod
+    def required_packages(cls) -> List[Text]:
+        """Lists required dependencies (see parent class for full docstring)."""
+        return ["spacy"]
 
-    def preprocess_text(self, text: Optional[Text]) -> Text:
-        """Processes the text before it is handled by spaCy."""
+    @classmethod
+    def create(
+        cls,
+        config: Dict[Text, Any],
+        model_storage: ModelStorage,
+        resource: Resource,
+        execution_context: ExecutionContext,
+    ) -> SpacyPreprocessor:
+        """Creates component for training see parent class for full docstring)."""
+        return cls(config)
+
+    def _doc_for_text(self, model: Language, text: Text) -> Doc:
+        """Makes a SpaCy doc object from a string of text."""
+        return model(self._preprocess_text(text))
+
+    def _preprocess_text(self, text: Optional[Text]) -> Text:
+        """Processes the text before it is handled by SpaCy."""
         if text is None:
             # converted to empty string so that it can still be passed to spacy.
             # Another option could be to neglect tokenization of the attribute of
             # this example, but since we are processing in batch mode, it would
             # get complex to collect all processed and neglected examples.
             text = ""
-        if self.component_config.get("case_sensitive"):
+        if self._config.get("case_sensitive"):
             return text
         else:
             return text.lower()
 
-    def get_text(self, example: Dict[Text, Any], attribute: Text) -> Text:
-
-        return self.preprocess_text(example.get(attribute))
+    def _get_text(self, example: Dict[Text, Any], attribute: Text) -> Text:
+        return self._preprocess_text(example.get(attribute))
 
     @staticmethod
-    def merge_content_lists(
+    def _merge_content_lists(
         indexed_training_samples: List[Tuple[int, Text]],
-        doc_lists: List[Tuple[int, "Doc"]],
-    ) -> List[Tuple[int, "Doc"]]:
+        doc_lists: List[Tuple[int, Doc]],
+    ) -> List[Tuple[int, Doc]]:
         """Merge lists with processed Docs back into their original order."""
-
         dct = dict(indexed_training_samples)
         dct.update(dict(doc_lists))
```

Comment:
Actually don't need to convert `doc_lists` to `dict` as `dict.update()` accepts a list of tuples (see [here](https://docs.python.org/3/library/stdtypes.html?highlight=dict#dict.update))